server/src/main/java/org/elasticsearch/transport/TcpTransport.java

```diff
@@ -1021,10 +987,13 @@ private BytesReference buildMessage(long requestId, byte status, Version nodeVer
      */
     public void inboundMessage(TcpChannel channel, BytesReference message) {
         try {
+            channel.getStats().markRead();
             transportLogger.logInboundMessage(channel, message);
             // Message length of 0 is a ping
             if (message.length() != 0) {
                 messageReceived(message, channel);
+            } else {
+                keepAlive.receiveKeepAlive(channel);
```

Comment:
this took me a moment to understand that we only do anything if we are a server channel in this case. Can you leave a comment here to explain that we only send back something if we are a server channel to ensure we don't ping back and forth all the time. The ping is always driven by the client essentially, right?

Reply:
I added the comment in ` keepAlive.receiveKeepAlive(channel)`.